Code/Framework/AzCore/AzCore/Console/ConsoleFunctor.h

```diff
@@ -127,9 +139,13 @@ namespace AZ
     {
     public:
 
-        using MemberFunctorSignature = void(_TYPE::*)(const ConsoleCommandContainer&);
+        using MemberFunctorSignature = typename ConsoleCommandMemberFunctorSignature<_TYPE>::type;
         using RawFunctorSignature = void(*)(_TYPE&, const ConsoleCommandContainer&);
-        using FunctorUnion = AZStd::variant<RawFunctorSignature, MemberFunctorSignature>;
+
+        using FunctorUnion = AZStd::conditional_t<
```

Comment:
suggestion: Maybe a comment here would be useful to explain what this alias is doing and why it's needed

Reply:
Yeah, that could be done.
The FunctorUnion alias is avoiding the issue with two of the exact types being added as alternatives to a variant.
That causes ambiguity errors when initializing the variants without using the explicit index of the alternative to use.

So the solution is to collapse the variant to a single alternative as there isn't a need to have two alternatives in this case.

Reply:
I see interesting, yeah as it's a bit of a strange case a comment could be helpful

Reply:
yep, added a comment, thanks!